### DISCUSSION TITLE
How to ignore the timeout error in loading tiles?

### DISCUSSION THREAD

**Discussion by stevendai11** [openseadragon/openseadragon#2045] [Q&A]
Hi,
During loading the tiles using AddTiledImage with a url, after loading many images like 16-20, I got loading time out error from previous images, like image number 1-5(exceeding 30000ms). While it doesn't prevents me to load new images, it does pause for quite a while dumping/logging the errors/messages. I was wondering if there is a way to let osd simply ignore the timeout errors and not log the messages so the new image could be loaded without delay?
Thanks a lot for your help.
Steven
P.S. I have tried to set timeout value but I am afraid it eventually accumalates and makes the problem worse.

**Comment by iangilman** [comment]
How big are these images? Are they tiled? Loading 20 images shouldn't cause timeouts like that. 

I'm not sure where the pause is coming from, but I doubt it's due to the logging. There's no way to suppress just that one message, but you can suppress all OpenSeadragon logging like so:

```
OpenSeadragon.console.log = function() {};
```

You can test that and see if the pause goes away. 

At any rate, I think the deeper issue is that the images are taking a long time to load... there may be some way to optimize that.

**Reply by stevendai11** [reply]
Thanks for the reply.
These files are around 2G each, medical related files with say 15 layers and l load layer by layer now. I tried loading layers in a batch but too many such errors so I end up using layer per click or request. It still shows error but much less. I assume it is too much stress to the server but I am limited to upgrading the server. So just want to try a workaround.
I will add your suggestion shortly to see if it makes any difference.
Thanks again for your help.

**Reply by iangilman** [reply]
Those files are quite large! You should definitely convert them into a tiled pyramid format ahead of time... that will make a huge difference in the load time! 

There are a number of options for formats and converters: 

http://openseadragon.github.io/examples/creating-zooming-images/

I recommend using VIPS to convert to DZI.

**Reply by stevendai11** [reply]
Thanks a lot again.
We are using osd AddTiledImage with Sources to fetch the tiled images from the server which provides processed tiled images.
I tried the console not logging modification. Unfortunately we are using typed version of osd which only has limited access to the osd settings where console is not one of them. When the functionality is available in the newer version of typed osd, I will certainly give a try.
Steven

**Reply by msalsbery** [reply]
> I tried the console not logging modification. Unfortunately we are using typed version of osd which only has limited access to the osd settings where console is not one of them. 

Can you use a plugin? https://github.com/openseadragon-imaging/openseadragon-consolehook

**Reply by iangilman** [reply]
So you are already tiling the images? Interesting. Can you share a link to a repro of the issue? If not, can you share a screenshot of your "network" inspector tab while it loads?

**Comment by stevendai11** [comment]
![image](https://user-images.githubusercontent.com/74571326/136083490-7a139b9a-47cc-48c4-8cc3-bf2685b2bf40.png)
THis is my VS code logging. My server ignores the error if it fails (doesn't have enough time) to process the tile request in time.
The angular client just times out.

**Reply by stevendai11** [reply]
@msalsbery 
no sure if it's necessary for my project since I have already limited the lay to only 1 per image for now.
But will try. 
Thanks a lot.
Steven

**Comment by stevendai11** [comment]
@msalsbery Hi,
Just tried the console plugin. Somehow it causes my osd viewer not working correctly. Nothing is logged including my own code in the module.
I did log.log = function(){};
I am using angular 12 with Typed osd version 0.9.
Thanks.

**Reply by msalsbery** [reply]
> @msalsbery Hi,
> Just tried the console plugin. Somehow it causes my osd viewer not working correctly. Nothing is logged including my own code in the module.
> I did log.log = function(){};
> I am using angular 12 with Typed osd version 0.9.
> Thanks.

Does it break the viewer or just stop logging?

I don’t know how much the plugin can help you, but thought it might be a way to turn off logging at least.

To log and/or prevent default logging with the plugin you need to add replacement logging functions …

[Documentation](https://openseadragon-imaging.github.io/docs/openseadragon-consolehook/index.html)

[Example code, around line 121](https://github.com/openseadragon-imaging/openseadragon-imaging/blob/master/site.github.io/src/page-consolehook/OsdViewer.js)

**Comment by stevendai11** [comment]
In my previous implementation,Log is the plug-in’ log. I could not find addConsoleHandler in the viewer object.Again I am not using vanilla js but typescript version’s osd. So I am not sure if your plugin works in my case.But thanks anyway.Steven 


Sent from Yahoo Mail for iPhone


On Tuesday, October 5, 2021, 1:08 PM, Mark Salsbery ***@***.***> wrote:





@msalsbery Hi,
Just tried the console plugin. Somehow it causes my osd viewer not working correctly. Nothing is logged including my own code in the module.
I did log.log = function(){};
I am using angular 12 with Typed osd version 0.9.
Thanks.


Does it break the viewer or just stop logging?

I don’t know how much the plugin can help you, but thought it might be a way to turn off logging at least.

To log with the plugin you need to add replacement logging functions …

Documentation

Example code, around line 121

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by iangilman** [comment]
@stevendai11 thank you for the screenshot of your console. I was actually hoping for a screenshot of the network tab of your browser developer inspector. That would show timing information, as well as all of your file requests, not just the failures. I just want to see the images portion of that. Here's an example from my Chrome:

<img width="850" alt="image" src="https://user-images.githubusercontent.com/338989/136262052-0f1d1b18-b01b-4ab4-8ae1-35236ff55a08.png">

The server you're using, is it tiling the images dynamically as you request them, or does it already have them pre-tiled?

**Reply by stevendai11** [reply]
Thanks Ian.
Sorry for the delay since I was tied up with other tasks.
Here is the snapshot.
![good_osd](https://user-images.githubusercontent.com/74571326/136583117-ca7fdc34-37d4-4c25-9447-12867ed23b16.PNG)
We are using osd (typed version) through Angular via CEFSharp. So it's kind of localhost etc and the network speed is quite fast.
When the server has troubling catching up, it just return an ingore status through http calls.
I am having another issue which osd when zooming from 20x to 40x, the image is blurred. I will ask a separate question and post the link below. If you or anyone could help, please have a look. Thanks again.

**Comment by stevendai11** [comment]
Hi Ian,
this is the link to my just posted question.
Thanks.
https://github.com/openseadragon/openseadragon/discussions/2048

**Comment by iangilman** [comment]
Thank you for sharing your network screenshot. It looks like every tile is taking 17 seconds to send? I would say that's definitely the problem! There must be something you can do to speed up the server! Are you generating the tiles in response to the request, or are they already pre-built? 

Also, I see you're using BMPs. Keep in mind that not all browsers support that format; JPG would be better for browser support (and file size).

**Comment by stevendai11** [comment]
Thanks for spotting the time. Actually that's at the very end of switching among images and layers to invoke intensive server loading and generating timeout errors. So I assume it does have waited 17 seconds or even to 30s.
But here is what happens a the begining of the ops
![image](https://user-images.githubusercontent.com/74571326/136844058-15ac7f78-456f-4827-94ae-2db9a7f61923.png)

**Comment by iangilman** [comment]
Cool. Those numbers look much better! 

I do think the problem lies in your server. Are you generating the tiles on the fly?

**Comment by stevendai11** [comment]
Yes.


Sent from Yahoo Mail for iPhone


On Tuesday, October 12, 2021, 11:22 AM, Ian Gilman ***@***.***> wrote:




Cool. Those numbers look much better!

I do think the problem lies in your server. Are you generating the tiles on the fly?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by iangilman** [comment]
I would suggest looking into some way to cache the tiles on the server, so you don't have to do the work to retile them on every request. It looks like it's already causing performance issues without caching, and it's only going to get more dramatic once you have more users.

**Comment by liveman1** [comment]
Hi, just as an idea….  I get performance issues using vpn. Might it also be a firewall issue?
